### PR TITLE
refactor local dev to simulate ingress

### PR DESCRIPTION
**Pull request by longrunningprocess**

### CODE REVIEW

File: frontend/vite.config.ts [hunk 1]
```diff
@@ -19,22 +19,4 @@ export default defineConfig({
     test: {
         include: ['src/**/*.{test,spec}.{js,ts}'],
     },
-    server: {
```
Comment (hahn-kev) [inline frontend/vite.config.ts:22]: can we please add this back? without this I can't run development locally unless I use docker for everything.

File: proxy/default.conf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+	location / {
+		proxy_pass http://ui:3000;
+	}
+
+	location /api {
+		proxy_pass http://lexbox-api:5158;
+	}
+
+	location /v1/traces {
```
Comment (hahn-kev) [inline proxy/default.conf:14]: here you have the location for traces as `/v1/traces` however in ingress it's just `/traces` shouldn't they be the same?

**Review by hahn-kev — COMMENTED** [review]
I'm fine with simulating ingress for local dev, but I still want to be able to run local dev without docker, which means that vite needs to be able to route requests to dotnet without going through docker.

File: proxy/default.conf (hunk 2 above)
Comment (longrunningprocess) [reply proxy/default.conf:14]: yeah, I was trying to troubleshoot that as well, I thought it was supposed to be `/traces` but I was actually seeing `/v1/traces`... more work to do there to get things right.

File: frontend/vite.config.ts (hunk 1 above)
Comment (longrunningprocess) [reply frontend/vite.config.ts:22]: maybe we could just have separate vite configs to give you what you want.

**Pull request merged**